### DISCUSSION TITLE
Default to reverse sort order?

### DISCUSSION THREAD

**Discussion by johnrix** [dbeaver/dbeaver#22917] [Q&A] [answered]
Is there any way to have DBeaver sort data in reverse PK order by default when opening tables in the data viewer? I am generally always interested in seeing the most recent records in a table when opening it up, rather than the earliest, so this would save me a lot of time if it were possible. I have searched through the preferences but not found any so far.

For reference, I am working with MySQL 8.x.

**Comment by LonwoLonwo** [comment]
No, DBeaver doesn't have such a feature.

The only helpful thing I can suggest is saving the ordering filter. After DBeaver or the table reopens - it will still have the correct order.

![2024-02-22 19_30_06-DBeaver Ultimate 23 3 5 - help_topic](https://github.com/dbeaver/dbeaver/assets/45152336/59a1c297-aa8d-4136-b3c1-e7665c023579)

**Reply by johnrix** [reply]
Thanks! That's pretty close to the same thing, so works for me!

**Comment by rosedo** [comment]
It's nice to be able to save these settings for a table. Of course different tables they have different structure.

It would still be really useful to have some checkbox to enable a default order by desc of the primary key.
Usually it's an id that increases over time and new records are almost always more relevant to us.

**Reply by johnrix** [reply]
After having been using this feature for a while now, I tend to agree with you. Although it remembers the sort order for each table that I save it for, having to do so for each individual table on each database gets very tedious, so it is not a perfect workaround by any means. Having a global default sort order for tables that have an id column would be a much simpler solution.

**Comment by ToreyShewmake** [comment, marked as answer]
This is now an option under Window > Preferences > Editor > Data Editor > Queries > Default Ordering

Found this post on Google right above the actual answer

**Reply by johnrix** [reply]
Fantastic - I never spotted this! Thanks for flagging it, and a big thanks to the dbeaver devs, this is a huge help!

**Comment by danikane** [comment]
Amazing, came here to say thanks!

**Comment by bartlomiejbaronlh** [comment]
Me too! Thanks :D
